### ISSUE TITLE
PlanCache: caching modifications on meta tables leads to crashing

### ISSUE THREAD

**Issue by Bensk1** [hyrise/hyrise#2049]

**Comment by mrks** [comment]
What's the status here? How can this be reproduced?

**Comment by Bensk1** [comment]
The bug is still present. For reproduction, build `hyriseConsole` and `hyriseTestPlugin`. In the console:

1. `INSERT INTO meta_plugins (name) VALUES ('PATH_TO/libhyriseTestPlugin.dylib');`
1. `DELETE FROM meta_plugins WHERE name = 'hyriseTestPlugin';`
1. `INSERT INTO meta_plugins (name) VALUES ('PATH_TO/libhyriseTestPlugin.dylib');` > BAM

I verified that 1. places a PQP in the cache and that 3. retrieves a PQP from the cache. Also, I quickly looked at sql_translator.cpp and in contrast to `_translate_delete()`, `_translate_update()` does not call `_translate_meta_table()` while `_cacheable` is also not set to `false`. I don't know if it would be sufficient to simply set it to `false` in `_translate_update()` (and write a test for it) or if the issue is more complicated. @dey4ss implemented this functionality IIRC. Maybe he can comment and/or have a look at it?

**Comment by dey4ss** [comment]
I'll have a look at it

**Comment by Bensk1** [comment]
Thanks!

**Comment by mrks** [comment]
Slightly unrelated to this PR, feel free to spin off into its own issue if it isn't done in ten lines.

Quick thought that I had when looking at #2104 and thinking back at how the meta table were generated many times. Should we add a test with a fake, instrumented meta table that tracks how often it is generated, query it, and verify that it was generated only once? Run the same pipeline again and it should be generated again.

**Comment by dey4ss** [comment]
> Should we add a test with a fake, instrumented meta table that tracks how often it is generated, query it, and verify that it was generated only once? Run the same pipeline again and it should be generated again.

That shouldn't be a problem to add as MetaMockTable already offers access counters. I can add a test here.
